include/glow/Quantization/Base/Base.h

```diff
@@ -98,6 +97,30 @@ inline float dequantize(eTy input, const TensorQuantizationParams &TQP) {
   return TQP.scale * ((int64_t)input - TQP.offset);
 }
 
+/// Converts floating point value to DestTy (quantized type) based on the
+/// quantization parameters \p scale and \p offset. If the dest type is int8_t,
+/// then an offset of 128 is substracted to convert to int8_t.
+template <class DestTy>
+inline DestTy quantizeWithFloatOffset(float input, float scale, float offset) {
+  uint8_t d = static_cast<uint8_t>((input - offset) / scale);
+  if (std::is_same<int8_t, DestTy>::value) {
+    d -= 128;
```

Comment:
Hmm... this could overflow the uint (which is ok), but then if the result is assigned to an int, it's also maybe UB.  Can you check this too?

Reply:
According to Integral conversions [in the spec](https://en.cppreference.com/w/cpp/language/implicit_conversion) this is "implementation-defined", but not undefined. It also passes all checks using UBSan locally/on CI.